%YAML:1.0
cameras:
    - {T_SC:
       [ 0.03374287,  0.00631998,  0.99941056,  0.10030527,
         0.99941876,  0.00464289, -0.03377251, -0.07156942,
        -0.00485359,  0.99996925, -0.00615964,  0.02466609,
          0.0, 0.0, 0.0, 1.0],
       image_dimension: [1600, 1100],
       distortion_coefficients: [ -0.076681019896292, 0.05597074463269014, -6.58597206142547e-05,  -0.00036397743435735637],
       distortion_type: radialtangential,
       focal_length: [875.8760546337106, 874.3320431510673], 
       principal_point: [795.6991969348003, 559.9630752248089]} 
    - {T_SC:
       [ 0.02118701,  0.00713449,  0.99975007,  0.10218905,
         0.99977542,  0.00032287, -0.02118985,  0.03740254,
        -0.00047397,  0.9999745,  -0.00712604, 0.02449153,
          0.0, 0.0, 0.0, 1.0],
       image_dimension: [1600, 1100],
       distortion_coefficients: [ -0.07998124641068913, 0.059126182968817026, -0.00012304198992984424, -0.0004879647198666465], 
       distortion_type: radialtangential,
       focal_length: [877.3653387528966, 876.1297548092742],
       principal_point: [794.1612517518113, 549.4142750061043]}  

camera_params:
    camera_rate: 20
    sigma_absolute_translation: 0.0
    sigma_absolute_orientation: 0.0
    sigma_c_relative_translation: 0.0 # below 1.0e-6 ruines numerics...
    sigma_c_relative_orientation: 0.0 # below 1.0e-6 ruines numerics...


imu_params:
    a_max: 176.0 # acceleration saturation [m/s^2]
    g_max: 7.8 # gyro saturation [rad/s]
    sigma_g_c: 0.0016017 # gyro noise density [rad/s/sqrt(Hz)] 0.00016017
    sigma_a_c: 0.0071376 # accelerometer noise density [m/s^2/sqrt(Hz)] 0.00071376
    sigma_bg: 0.03 # gyro bias prior [rad/s]
    sigma_ba: 0.1 # accelerometer bias prior [m/s^2]
    sigma_gw_c: 0.0000165 # gyro drift noise density [rad/s^s/sqrt(Hz)] 0.00000165
    sigma_aw_c: 0.0002874 # accelerometer drift noise density [m/s^2/sqrt(Hz)] 0.00002874
    tau: 3600.0 # reversion time constant, currently not in use [s]
    g: 9.81007 # Earth's acceleration due to gravity [m/s^2]
    a0: [ 0.0, 0.0, 0.0 ] # Accelerometer bias [m/s^2]
    imu_rate: 100
    # tranform Body-Sensor (IMU)
    T_BS:
        [1.0000, 0.0000, 0.0000, 0.0,
         0.0000, 1.0000, 0.0000, 0.0,
         0.0000, 0.0000, 1.0000, 0.0,
         0.0000, 0.0000, 0.0000, 1.0000]


#gyroscope_noise_density: 0.2e-3, gyroscope_random_walk: 2.8577e-006

# Estimator parameters
numKeyframes: 5 # number of keyframes in optimisation window
numImuFrames: 3 # number of frames linked by most recent nonlinear IMU error terms

# ceres optimization options
ceres_options:
    minIterations: 3   # minimum number of iterations always performed
    maxIterations: 10  # never do more than these, even if not converged
    timeLimit: 0.035   # [s] negative values will set the an unlimited time limit

# detection
detection_options:
    threshold: 40.0      # detection threshold. By default the uniformity radius in pixels
    octaves: 0           # number of octaves for detection. 0 means single-scale at highest resolution
    maxNoKeypoints: 400  # restrict to a maximum of this many keypoints per image (strongest ones)   #Sharmin

# delay of images [s]:
imageDelay: 0.0  # in case you are using a custom setup, you will have to calibrate this. 0 for the VISensor.

# display debug images?
displayImages: true  # displays debug video and keyframe matches. May be slow.
publishDebugImages: false  # publish debug image to topic ~debug_image_x

# use direct driver
useDriver: false

# some options for how and what to publish -- optional in ROS-free version
publishing_options:
    publish_rate: 100                  # rate at which odometry updates are published only works properly if imu_rate/publish_rate is an integer!!
    publishLandmarks: true             # select, if you want to publish landmarks at all
    landmarkQualityThreshold: 1.0e-2   # landmark with lower quality will not be published
    maximumLandmarkQuality: 0.05       # landmark with higher quality will be published with the maximum colour intensity
    maxPathLength: 20                  # maximum length of the published path
    publishImuPropagatedState: false    # Should the state that is propagated with IMU messages be published? Or just the optimized ones?
    # provide custom World frame Wc
    T_Wc_W:
        [1.0000, 0.0000, 0.0000, 0.0000,
         0.0000, 1.0000, 0.0000, 0.0000,
         0.0000, 0.0000, 1.0000, 0.0000,
         0.0000, 0.0000, 0.0000, 1.0000]
    trackedBodyFrame: B                # B or S, the frame of reference that will be expressed relative to the selected worldFrame
    velocitiesFrame: Wc                # Wc, B or S,  the frames in which the velocities of the selected trackedBodyFrame will be expressed in

#Sharmin
isSonarUsed: false
isDepthUsed: false
isRelocalization: true
histogramMethod: "NONE"     #NONE, HISTOGRAM, CLAHE
claheClipLimit: 2.0
claheTilesGridSize: 4
resizeFactor: 0.5  # by default 1.0. set this value if you want to resize the original image by this factor. You DO NOT need to change camera/projection matrix, those will be updated accordingly. 

